app/incometax/business/controllers/BusinessAccountingPeriodDateController.scala

```diff
@@ -74,25 +76,28 @@ class BusinessAccountingPeriodDateController @Inject()(val baseConfig: BaseContr
           editMatch = editMatch
         ))),
         accountingPeriod => {
-          if(accountingPeriodService.checkEligibleAccountingPeriod(
-            accountingPeriod.startDate.toLocalDate,
-            accountingPeriod.endDate.toLocalDate)) {
+          keystoreService.fetchRentUkProperty() flatMap { ukProperty =>
+            if (accountingPeriodService.checkEligibleAccountingPeriod(
+              accountingPeriod.startDate.toLocalDate,
+              accountingPeriod.endDate.toLocalDate,
+              ukProperty.contains(RentUkPropertyModel(Yes, Some(No))))) {
```

Comment:
```suggestion
              ukProperty.exists(_.rentUkProperty == Yes))) {
```
Might be more readable checking this way